### PR TITLE
Day-6

### PR DESCRIPTION
**Pull request by madhusdhnn**

### CODE REVIEW

File: src/main/java/org/example/list/SinglyLinkedList.java [hunk 1]
```diff
@@ -266,6 +278,210 @@ public void uniqueSorted() {
...
+                list1 = list1.next;
+            } else {
+                merged.next = list2;
+                list2 = list2.next;
+            }
+            merged = merged.next;
+        }
+
+        if (list1 != null) {
+            if (list1.next == null) {
```
Comment (venkatyuva) [inline src/main/java/org/example/list/SinglyLinkedList.java:438]: Why are we doing this here?

File: src/main/java/org/example/list/SinglyLinkedList.java [hunk 2]
```diff
@@ -266,6 +278,210 @@ public void uniqueSorted() {
...
+            if (list2.next == null) {
+                this.tail = list2;
+            }
+            merged.next = list2;
+        }
+
+        return dummy.next;
+    }
+
+    public void partition(int pivot) {
```
Comment (venkatyuva) [inline src/main/java/org/example/list/SinglyLinkedList.java:454]: Is this code working for all the test cases in gfg/leetcode?

**Review by venkatyuva — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/list/SinglyLinkedList.java (hunk 2 above)
Comment (madhusdhnn) [reply src/main/java/org/example/list/SinglyLinkedList.java]: Yes

File: src/main/java/org/example/list/SinglyLinkedList.java (hunk 1 above)
Comment (madhusdhnn) [reply src/main/java/org/example/list/SinglyLinkedList.java:496]: Customization of `tail` node in this code. It is not required in the actual submission code

**Pull request merged**